packages/drivers/odsp-driver/api-report/odsp-driver.api.md

```diff
@@ -158,6 +159,14 @@ export interface IPersistedFileCache {
     removeEntries(): Promise<void>;
 }
 
+// @alpha (undocumented)
+export interface IPrefetchPartialSnapshotWithContents extends IPartialSnapshotWithContents {
```

Comment:
How much should this layer care if it's partial snapshot, and if it has content?
Maybe it's not that bad here, but if every single place that passes through snapshots needs to now have special code to deal with these new cases (and define new interfaces), then our code would be littered with these constructs. Consider earlier advice to make IPartialSnapshotWithContents castable to ISnapshotTree and most of the signatures just using ISnapshotTree only (and leave code doing sniffing for more concrete interface)

Reply:
Made IPartialSnapshotWithContents inherit from ISnapshotTree but lets keep IPartialSnapshotWithContents and ISnapshotContents separate as it would be complicated to mix these 3 types. Also, ISnapshotConents is just localized to odsp driver. So, it would have minimal impact.

However, IPrefetchPartialSnapshotWithContents represents the snapshot which is prefetched through the functionality provided by odsp driver.